src/main/scala/scalafix/sbt/ScalafixEnable.scala

```diff
@@ -2,66 +2,184 @@ package scalafix.sbt
+      partialVersion <- CrossVersion.partialVersion(scalaV).toList
+      maybeRecommendedSemanticdbScalacV <-
+        maybeRecommendedSemanticdbScalacScalaVersion.lift(partialVersion).toList
+      scalafixResolvers0 <- (p / scalafixResolvers)
+        .get(extracted.structure.data)
+        .toList
+      semanticdbCompilerPlugin0 <- (p / semanticdbCompilerPlugin)
+        .get(extracted.structure.data)
+        .toList
+    } yield CompatibleProject(
+      p,
+      VersionNumber(scalaV),
+      semanticdbCompilerPlugin0,
+      scalafixResolvers0,
+      maybeRecommendedSemanticdbScalacV
+    )
+
+  private case class CompatibleProject(
+      ref: ProjectRef,
+      scalaVersion0: VersionNumber,
+      semanticdbCompilerPlugin0: ModuleID,
+      scalafixResolvers0: Seq[Repository],
+      maybeRecommendedSemanticdbScalacScalaV: Option[VersionNumber]
+  )
 
   lazy val command = Command.command(
     "scalafixEnable",
-    briefHelp = "Configure SemanticdbPlugin for scalafix.",
-    detail = """1. set semanticdbEnabled & semanticdbVersion
-      |2. conditionally sets scalaVersion when support is not built-in in the compiler""".stripMargin
+    briefHelp =
+      "Configure SemanticdbPlugin for scalafix on supported projects.",
+    detail = """1. set semanticdbEnabled := true
+      |2. for scala 2.x,
+      |  - set semanticdbCompilerPlugin to the scalameta version tracked by scalafix if available for scalaVersion, 
+      |  - otherwise set semanticdbCompilerPlugin to a compatible version available for scalaVersion,
+      |  - otherwise force scalaVersion to the latest version supported by the scalameta version tracked by scalafix.""".stripMargin
   ) { s =>
     val extracted = Project.extract(s)
     val scalacOptionsSettings = Seq(Compile, Test).flatMap(
       inConfig(_)(ScalafixPlugin.relaxScalacOptionsConfigSettings)
     )
     val settings = for {
-      (p, maybeFullVersion) <- collectProjects(
-        extracted,
-        maybeSemanticdbScalacFullScalaVersion
-      )
-      enableSemanticdbPlugin <- maybeFullVersion.toList.map { fullVersion =>
-        scalaVersion := fullVersion,
-      } :+ (semanticdbEnabled := true)
+      project <- collectProjects(extracted)
+      enableSemanticdbPlugin <-
+        project.maybeRecommendedSemanticdbScalacScalaV.toList
+          .flatMap { recommendedSemanticdbScalacScalaV =>
+
+            import scalafix.internal.sbt.Implicits._
+            val semanticdbScalacModule =
+              coursierapi.Dependency
+                .parse(
+                  project.semanticdbCompilerPlugin0.asCoursierCoordinates,
+                  coursierapi.ScalaVersion.of(project.scalaVersion0.toString)
+                )
+                .getModule
+            val recommendedSemanticdbV =
+              VersionNumber(BuildInfo.scalametaVersion)
+            val compatibleSemanticdbVs = Try(
+              coursierapi.Versions.create
```

Comment:
To my knowledge, this is the best Java coursier API to achieve something like suggested in https://github.com/scalacenter/scalafix/issues/1146#issuecomment-1039971557